src/main/java/com/iota/iri/zmq/MessageQ.java

```diff
@@ -29,12 +32,19 @@ public MessageQ(int port, String ipc, int nthreads, boolean enabled) {
     }
 
     public void publish(String message, Object... objects) {
-        if(enabled) {
-            publisher.send(String.format(message, objects));
+        if (enabled) {
+            String toSend = String.format(message, objects);
+            publisherService.submit(() -> publisher.send(toSend));
         }
     }
 
-    public void shutdown () {
+    public void shutdown() {
+        publisherService.shutdown();
+        try {
+            publisherService.awaitTermination(5, TimeUnit.SECONDS);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
         publisher.close();
```

Comment:
Since there is a `try` I find it more readable to put the last two lines in a `finally` block.